### PR TITLE
Merge user config branch

### PR DESCRIPTION
**Pull request by nbennett320**
Allows key/control remapping

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,4 +10,6 @@ a.out
 roms/
 third_party/fmt
 bios.bin
-*.sav
\ No newline at end of file
+*.sav
+*.swp
```
Comment (samkravitz) [inline .gitignore:14]: This is a vim thing right? We don't need to commit it in .gitignore

File: Makefile [hunk 2]
```diff
@@ -6,10 +6,10 @@ SOURCE = src
 INCLUDE = include
 BACKUPDIR = $(INCLUDE)/backup
 OBJ = \
+	config.o \
```
Comment (samkravitz) [inline Makefile:9]: Thank you for fixing alphabetical order here

File: .gitignore (hunk 1 above)
Comment (nbennett320) [reply .gitignore:14]: .sav files are GBA save files, but .swp are vim

Comment (nbennett320) [reply .gitignore]: i removed .swp

**Pull request merged**